statistics/statistics_test.go

```diff
@@ -64,141 +65,151 @@ func TestStatisticsProcessor(t *testing.T) {
 func TestStatisticsProcessorMetrics(t *testing.T) {
 	Convey("Statistics Processor tests", t, func() {
 		metrics := make([]plugin.Metric, 10)
-		data := [10]float64{5, 12, 7, 9, 33, 53, 24, 16, 18, 1}
-
-		time := [10]time.Time{time.Now().Add(12 * time.Hour),
-			time.Now().Add(22 * time.Hour),
-			time.Now().Add(9 * time.Hour),
+		var stats []plugin.Metric
+		var err error
+		data := []float64{33, 53, 24, 16, 18, 1, 7, 9, 5, 12}
+		time := [10]time.Time{time.Now().Add(22 * time.Hour),
+			time.Now().Add(12 * time.Hour),
 			time.Now().Add(10 * time.Hour),
-			time.Now().Add(1 * time.Hour),
-			time.Now().Add(2 * time.Hour),
-			time.Now().Add(3 * time.Hour),
-			time.Now().Add(5 * time.Hour),
-			time.Now().Add(6 * time.Hour),
+			time.Now().Add(9 * time.Hour),
 			time.Now().Add(7 * time.Hour),
+			time.Now().Add(6 * time.Hour),
+			time.Now().Add(5 * time.Hour),
+			time.Now().Add(3 * time.Hour),
+			time.Now().Add(2 * time.Hour),
+			time.Now().Add(1 * time.Hour),
 		}
+
 		config := plugin.Config{}
-		config["SlidingWindowLength"] = int64(5)
+		config["slidingWindowLength"] = int64(5)
+		config["slidingFactor"] = int64(1)
+		config["statistics"] = strings.Join(statList, ",")
 
 		empty := []float64{}
+		modes := [][]float64{[]float64{33}, empty, empty, empty, empty, empty, empty, empty, empty, empty}
+
+		expected := make(map[string][]float64)
+		expected[sum] = []float64{33, 86, 110, 126, 144, 112, 66, 51, 40, 34}
+		expected[count] = []float64{1, 2, 3, 4, 5, 5, 5, 5, 5, 5}
+		expected[minimum] = []float64{33, 33, 24, 16, 16, 1, 1, 1, 1, 1}
+		expected[maximum] = []float64{33, 53, 53, 53, 53, 53, 24, 18, 18, 12}
+		expected[mean] = []float64{33, 43, 36.66666667, 31.5, 28.8, 22.4, 13.2, 10.2, 8, 6.8}
+		expected[median] = []float64{33, 43, 33, 28.5, 24, 18, 16, 9, 7, 7}
+		expected[standarddeviation] = []float64{0, 10, 12.120, 13.793, 13.467, 17.072, 8.183, 6.177, 5.657, 3.709}
+		expected[variance] = []float64{0, 100, 146.889, 190.25, 181.36, 291.44, 66.96, 38.16, 32, 13.76}
+		expected[ninetyninthpercentile] = []float64{33, 53, 53, 53, 53, 53, 24, 18, 18, 12}
+		expected[ninetyfifthpercentile] = []float64{33, 53, 53, 53, 53, 53, 24, 18, 18, 12}
+		expected[secondpercentile] = []float64{33, 33, 24, 16, 16, 1, 1, 1, 1, 1}
+		expected[ninthpercentile] = []float64{33, 33, 24, 16, 16, 1, 1, 1, 1, 1}
+		expected[twentyfifthpercentile] = []float64{33, 33, 24, 16, 18, 16, 7, 7, 5, 5}
+		expected[seventyfifthpercentile] = []float64{33, 53, 53, 33, 33, 24, 18, 16, 9, 9}
+		expected[ninetyfirstpercentile] = []float64{33, 53, 53, 53, 53, 53, 24, 18, 18, 12}
+		expected[ninetyeighthpercentile] = []float64{33, 53, 53, 53, 53, 53, 24, 18, 18, 12}
+		expected[kurtosis] = []float64{math.NaN(), math.NaN(), math.NaN(), 1.8964, 2.337, 2.563, 1.6874, 1.6624, 2.4383, 2.0035}
```

Comment:
I see, NaN [can't be marshalled](
https://play.golang.org/p/TfW_skKFtK), what if we don't create metric when the value is Nan?